lib/lint.ml

```diff
@@ -437,67 +517,79 @@ module Lint = struct
-      | UnmatchedVersion value ->
-          Fmt.str "Error in %s: The field 'version' that doesn't match its context. \
-                   Field 'version' has value '%s' but was expected of value '%s'."
-            pkg
-            (OpamPackage.Version.to_string value)
-            (OpamPackage.Version.to_string (OpamPackage.version package))
-      | DubiousDuneSubst ->
-          Fmt.str "Warning in %s: Dubious use of 'dune subst'. \
-                   'dune subst' should always only be called with {dev} (i.e. [\"dune\" \"subst\"] {dev}) \
-                   If your opam file has been autogenerated by dune, you need to upgrade your dune-project \
-                   to at least (lang dune 2.7)."
-            pkg
-      | DuneProjectMissing ->
-          Fmt.str "Warning in %s: The package seems to use dune but the dune-project file is missing." pkg
-      | DuneConstraintMissing ->
-          Fmt.str "Warning in %s: The package has a dune-project file but no explicit dependency on dune was found." pkg
-      | DuneIsBuild ->
-          Fmt.str "Warning in %s: The package tagged dune as a build dependency. \
-                   Due to a bug in dune (https://github.com/ocaml/dune/issues/2147) this should never be the case. \
-                   Please remove the {build} tag from its filter."
-            pkg
-      | BadDuneConstraint (dep, ver) ->
-          Fmt.str "Error in %s: Your dune-project file indicates that this package requires at least dune %s \
-                   but your opam file only requires dune >= %s. Please check which requirement is the right one, and fix the other."
-            pkg ver dep
-      | UnexpectedFile file ->
-          Fmt.str "Error in %s: Unexpected file in %s/files/%s" pkg (Check.path_from_pkg package) file
-      | ForbiddenPerm file ->
-          Fmt.str
-            "Error in %s: Forbidden permission for file %s/%s. All files should have permissions 644."
-            pkg (Check.path_from_pkg package) file
-      | OpamLint warn ->
-          let warn = OpamFileTools.warns_to_string [warn] in
-          Fmt.str "Error in %s: %s" pkg warn
-      | FailedToDownload msg ->
-          Fmt.str "Error in %s: Failed to download the archive. Details: %s" pkg msg
-      | NameCollision other_pkg ->
-          Fmt.str "Warning in %s: Possible name collision with package '%s'" pkg other_pkg
-      | WeakChecksum msg ->
-        Fmt.str "Error in %s: Weak checksum algorithm(s) provided. Please use SHA-256 or SHA-512. Details: %s" pkg msg
-      | PinDepends ->
-        Fmt.str "Error in %s: pin-depends present. This is not allowed in the opam-repository." pkg
-    )
+  let msg_of_prefix_conflict_class_mismatch ~pkg = function
+    | WrongPrefix (conflict_class, required_prefix) ->
+      Fmt.str
+        "Error in %s: package with conflict class '%s' requires name prefix '%s'"
+        pkg conflict_class required_prefix
+    | WrongConflictClass (prefix, required_conflict_class) ->
```

Comment:
The names used in these patters would be nice choices for the record field names on the type, if you chose to go with those.